bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/RefreshCapability.java

```diff
@@ -86,27 +87,26 @@ private void proceedWithUpdate() {
             if (probing()) {
                 dataTimeStamp0 = Instant.MIN;
             }
-        } else if (refreshConfigured) {
-            delay = dataValidity.getSeconds();
         } else {
-            delay = (probing() ? PROBING_INTERVAL : dataValidity.minus(dataAge()).plus(DEFAULT_DELAY)).toSeconds();
+            delay = refreshConfigured ? dataValidity.getSeconds()
+                    : (probing() ? PROBING_INTERVAL : dataValidity.minus(dataAge()).plus(DEFAULT_DELAY)).toSeconds();
         }
         delay = delay < 2 ? PROBING_INTERVAL.toSeconds() : delay;
-        logger.debug("Module refreshed, next one in {} s", delay);
+        logger.debug("Module refreshed, next one in {}s", delay);
         freeJobAndReschedule(delay);
     }
 
     @Override
     protected void updateNAThing(NAThing newData) {
         super.updateNAThing(newData);
-        newData.getLastSeen().ifPresent(timestamp -> {
-            Instant tsInstant = timestamp.toInstant();
+        newData.getLastSeen().map(ZonedDateTime::toInstant).ifPresent(tsInstant -> {
             if (probing()) {
                 if (Instant.MIN.equals(dataTimeStamp0)) {
                     dataTimeStamp0 = tsInstant;
                     logger.debug("First data timestamp is {}", dataTimeStamp0);
                 } else if (tsInstant.isAfter(dataTimeStamp0)) {
                     dataValidity = Duration.between(dataTimeStamp0, tsInstant);
+                    refreshConfigured = true;
```

Comment:
I am just surprised. Why was it working for us before that change ?

Reply:
I observed that it worked only once - after that it went back to default 120s

Reply:
I believe you